lib/cooked_post_processor.rb

```diff
@@ -85,6 +86,29 @@ def post_process_quotes
     end
   end
 
+  def removed_direct_reply_full_quotes
+    return if @post.post_number == 1
+
+    num_quotes = @post.raw.scan(/\[quote[^\]]*\]/).size
+    return if num_quotes != 1
+
+    prev = @post.topic.posts.order('post_number desc').where('post_number < ? AND not hidden AND posts.deleted_at IS NULL', @post.post_number).limit(1).pluck(:raw).first
+    return if !prev
+
+    new_raw = @post.raw.gsub(/\[quote[^\]]*\]\s*#{Regexp.quote(prev.strip)}\s*\[\/quote\]/, '')
```

Comment:
I know we add the `full=true` attributes on the quote when they're full quotes. I don't remember how we check it though (if we do it at all).

Reply:
`full=true` attribute is only sometimes added. I noticed this when quoting bigger posts (i.e. including images).